homeassistant/components/tankerkoenig/binary_sensor.py

```diff
@@ -0,0 +1,80 @@
+"""Tankerkoenig binary sensor integration."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ID, ATTR_LATITUDE, ATTR_LONGITUDE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import TankerkoenigDataUpdateCoordinator
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the tankerkoenig sensors."""
+
+    coordinator: TankerkoenigDataUpdateCoordinator = hass.data[DOMAIN][entry.unique_id]
+
+    stations = coordinator.stations.values()
+    entities = []
+    for station in stations:
+        sensor = StationOpenBinarySensorEntity(
+            station,
+            coordinator,
+            coordinator.show_on_map,
+        )
+        entities.append(sensor)
+    _LOGGER.debug("Added sensors %s", entities)
+
+    async_add_entities(entities)
+
+
+class StationOpenBinarySensorEntity(CoordinatorEntity, BinarySensorEntity):
+    """Shows if a station is open or closed."""
+
+    _attr_device_class = BinarySensorDeviceClass.DOOR
+
+    def __init__(
+        self,
+        station: dict,
+        coordinator: TankerkoenigDataUpdateCoordinator,
+        show_on_map: bool,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._station_id = station["id"]
+        self._attr_name = (
+            f"{station['brand']} {station['street']} {station['houseNumber']} status"
```

Comment:
We might want to define a method to create the names, so that if we need to change that for both sensors (or future ones), a single place to modify is needed. Otherwise we might create inconsistencies

Reply:
I've already on my list, to create a kind of bases entity which defines at least the `device_info`, "base name" and some other shared used attributes